ci-scripts/teravm/fabfile.py

```diff
@@ -120,8 +120,34 @@ def upgrade_teravm(
 
     return hash
 
+def get_agw(setup):
+    version = None
+    """
+    Upgrade teravm agw to image with the given hash.
+    hash: a hash to identify what version from APT to use for upgrading.
+    If not hash provided or "latest" is passed, it will install latest
+    on the repository.
 
-def upgrade_teravm_agw(setup, hash, key_filename=DEFAULT_KEY_FILENAME):
+    key_filename: path to where the private key is for authorized-key based
+    ssh. The public key counterpart needs to in the authorized_keys file on
+    the remote host. If empty file name is passed, password-based ssh will
+    work instead. This can be used if the script is run manually.
+    """
+
+    fastprint("Upgrade teraVM AGW to %s")
+    _setup_env_agw("magma", VM_IP_MAP[setup]["gateway"])
+    err = _set_magma_apt_repo()
+    if err:
+        return err
+    sudo("apt update")
+    fastprint("Get latest version\n")
+    # Get the whole version string containing that hash and 'apt install' it
+    with settings(abort_exception=FabricException):
+	version = sudo ("apt-cache madison magma | awk 'NR==1{{print substr ($3,1)}}'")
```

Comment:
is this getting the latest, or the second to the latest version?

Maybe it would be good to add an argument `offset` to `get_agw(setup, offset)`. If offset is 0 it will get the latest, if it is 1 it will get the one before the latest... and so on